### PR TITLE
Rewrite Wheatley integration

### PR DESCRIPTION
**Pull request by kneasle**
Guess what - it's finally happened :tada:!

I've reimplemented the interface between RR and Wheatley so that hopefully zombie Wheatleys and multiple Wheatleys in a tower are both impossible.  These changes can't be easily tested until kneasle/wheatley#137 is merged, but I'll do that in the morning.

### CODE REVIEW

**Comment by BrynMarieR** [comment]
Thanks for doing this. We probably won't have time to look through carefully until Friday, but we're aware. Progress! We can send it out to the beta testers again soon. Also, let us know if you want to do your own testing on the server; we'll just need your public key.

**Comment by kneasle** [comment]
That's fine - I had sudden motivation to fix this yesterday, but no stress getting it merged quickly - Friday is fine.

I think I've already given you my public key, but I'll send it to you again once this gets merged.

**Comment by kneasle** [comment]
By the way, most of the code is one giant comment at the top of `wheatley.py`, which describes all the changes I've made in plain English.  The diffs of the code itself look a bit janky, but the code should be reasonably clean.

**Comment by lelandpaul** [comment]
Ben — I think this looks good! I'm going to merge it now and work on getting it up on the dev server so we can test it out some. I've just confirmed that your public key is on the dev server, so when you get a chance confirm that you can ssh in; I'll find a way to send you the other info you'll need.
